### PR TITLE
Add ApiClientWrapper methods

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Refactors our API Client Wrapper and moves it to its own module, with separate files for MLS and Identity methods
- Adds new Identity methods
- Adds mocks for Identity API and tests using those mocks

## Notes

- @insipx I borrowed some pieces of your pending PR that I needed here. Hopefully won't be too annoying with conflicts.
- Right now I'm being strict with serialization errors. We should think through how we really want to behave once this goes to prod. One person's bad update shouldn't break the entire group. And we should make sure this is safe to add new features like one-way associations later (since those don't change commit processing).

## Issues
- https://github.com/xmtp/libxmtp/issues/629

### CODE REVIEW

**Comment by neekolas** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/668?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#668** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/668?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#667** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xmtp_id/src/associations/test_utils.rs [hunk 1]
```diff
@@ -68,6 +79,35 @@ impl Signature for MockSignature {
...
+                    LegacyDelegatedSignatureProto {
+                        delegated_key: Some(LegacySignedPublicKeyProto {
+                            key_bytes: vec![0],
+                            signature: Some(LegacySignatureProto { union: None }),
+                        }),
+                        signature: Some(RecoverableEcdsaSignatureProto { bytes: vec![0] }),
+                    },
+                )),
+            },
+        }
```
Comment (37ng) [inline xmtp_id/src/associations/test_utils.rs:111]: thanks for adding these

File: xmtp_mls/src/api/identity.rs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+impl From<GetIdentityUpdatesV2Filter> for GetIdentityUpdatesV2RequestProto {
+    fn from(filter: GetIdentityUpdatesV2Filter) -> Self {
+        Self {
+            inbox_id: filter.inbox_id,
+            sequence_id: filter.sequence_id.unwrap_or(0),
+        }
+    }
+}
+
+pub struct IdentityUpdateV2 {
```
Comment (37ng) [inline xmtp_mls/src/api/identity.rs:33]: Could this be `IdentityUpdateV3` since it's for XMTP v3? Apparently it's second version of IdentityUpdate but I think it'd be more clear to unify the versioning across the board.

**Review by 37ng — APPROVED** [review]
A bit confused about the IdentityUpdate versioning

File: xmtp_mls/src/api/identity.rs (hunk 2 above)
Comment (neekolas) [reply xmtp_mls/src/api/identity.rs:33]: You're right, it's definitely confusing. We just already had a struct with `IdentityUpdate`. Maybe I can rename to `InboxUpdate`

**Review by richardhuaaa — APPROVED** [review]
Looks great

File: xmtp_mls/src/api/identity.rs (hunk 2 above)
Comment (neekolas) [reply xmtp_mls/src/api/identity.rs]: The good news is that once we're done this transition we can delete the old `IdentityUpdate` and rename everything

Comment (37ng) [reply xmtp_mls/src/api/identity.rs]: `InboxUpdate` sounds great

File: xmtp_id/src/associations/test_utils.rs (hunk 1 above)
Comment (neekolas) [reply xmtp_id/src/associations/test_utils.rs:111]: All credit goes to @insipx. I just stole them from his PR

**Comment by neekolas** [comment]
### Merge activity

* **Apr 17, 2:03 PM EDT**: @neekolas started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/xmtp/libxmtp/668).
